src/components/Referrals/ReferralRewardCard/ReferralRewardCard.spec.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import { Rarity } from '@dcl/schemas'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import type { ReferralTier } from '../../../modules/referrals/types'
+import { renderWithProviders } from '../../../tests/tests'
+import {
+  REFERRAL_REWARD_CARD_TEST_ID,
+  REFERRAL_REWARD_TIER_TEST_ID,
+  REFERRAL_REWARD_IMAGE_TEST_ID,
+  REFERRAL_REWARD_RARITY_TEST_ID,
+  REFERRAL_REWARD_SWAG_CONTAINER_TEST_ID,
+  REFERRAL_REWARD_DESCRIPTION_TEST_ID,
+  REFERRAL_REWARD_CLAIM_BUTTON_TEST_ID
+} from './constants'
+import { ReferralRewardCard } from './ReferralRewardCard'
+
+describe('ReferralRewardCard', () => {
+  let props: ReferralTier
+  let renderedComponent: ReturnType<typeof renderReferralRewardCard>
+
+  const renderReferralRewardCard = (props: ReferralTier) => {
+    return renderWithProviders(<ReferralRewardCard {...props} />)
+  }
+
+  beforeEach(() => {
+    props = {
+      tier: 1,
+      rarity: Rarity.COMMON,
+      completed: false,
+      image: 'test-image.png',
+      claim: false
+    }
+  })
+
+  describe('when rendering the component', () => {
+    beforeEach(() => {
+      renderedComponent = renderReferralRewardCard(props)
+    })
+
+    it('should render the card container', () => {
+      const { getByTestId } = renderedComponent
+      expect(getByTestId(REFERRAL_REWARD_CARD_TEST_ID)).toBeInTheDocument()
+    })
+
+    it('should display the correct tier', () => {
+      const { getByTestId } = renderedComponent
+      const tierElement = getByTestId(REFERRAL_REWARD_TIER_TEST_ID)
+      expect(tierElement).toHaveTextContent(t('referral_reward_card.tier', { count: props.tier }))
+    })
+
+    it('should display the image with correct alt text', () => {
+      const { getByTestId } = renderedComponent
+      const imageElement = getByTestId(REFERRAL_REWARD_IMAGE_TEST_ID)
+      expect(imageElement).toHaveAttribute('alt', t('referral_reward_card.image_alt'))
+      expect(imageElement).toHaveAttribute('src', props.image)
+    })
+
+    it('should display the rarity in uppercase', () => {
+      const { getByTestId } = renderedComponent
+      const rarityElement = getByTestId(REFERRAL_REWARD_RARITY_TEST_ID)
+      expect(rarityElement).toHaveTextContent(props.rarity.toUpperCase())
+    })
+
+    it('should display the unlock message when not completed', () => {
+      const { getByTestId } = renderedComponent
+      const descriptionElement = getByTestId(REFERRAL_REWARD_DESCRIPTION_TEST_ID)
+      expect(descriptionElement).toHaveTextContent(t('referral_reward_card.unlock'))
+    })
```

Comment:
What do you think about moving this test into the context of not being completed?